src/chimera/interfaces/telescope.py

```diff
@@ -417,6 +417,28 @@ def unparkComplete(self):
         successfully.
         """
 
+    def openCover(self):
```

Comment:
Yes, I do:

you must ask the class and do not test the proxy.

def hasFeature(self, feature/cls):
 ...

Cheers,

2015-10-23 15:23 GMT-02:00 William Schoenell <notifications@github.com>:

> In src/chimera/interfaces/telescope.py
> <https://github.com/astroufsc/chimera/pull/116#discussion_r42890653>:
>
> > @@ -417,6 +417,28 @@ def unparkComplete(self):
> >          successfully.
> >          """
> >
> > +    def openCover(self):
>
> Yeah... The idea is VERY good but Pyro does not let you do that... :-(
>
> from chimera.interfaces.telescope import Telescopefrom chimera.core.manager import Manager
> manager = Manager()
> telescope = manager.getProxy("127.0.0.1:7666/Telescope/0")print(type(telescope))issubclass(telescope, Telescope)
>
> gives you
>
> <class 'chimera.core.proxy.Proxy'>
> Traceback (most recent call last):
>   File "<string>", line 6, in <module>
> TypeError: issubclass() arg 1 must be a class
>
> Any ideas?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/astroufsc/chimera/pull/116/files#r42890653>.
>